app/assets/javascripts/discourse/app/components/about-page-user.gjs

```diff
@@ -0,0 +1,56 @@
+import Component from "@glimmer/component";
+import { or } from "truth-helpers";
+import avatar from "discourse/helpers/avatar";
+import { prioritizeNameInUx } from "discourse/lib/settings";
+import { userPath } from "discourse/lib/url";
+import i18n from "discourse-common/helpers/i18n";
+
+export default class AboutPageUser extends Component {
+  get prioritizeName() {
+    return prioritizeNameInUx(this.args.user.name);
+  }
+
+  <template>
+    <div data-username={{@user.username}} class="user-info small">
+      <div class="user-image">
+        <div class="user-image-inner">
+          <a
+            href={{userPath @user.username}}
+            data-user-card={{@user.username}}
+            aria-hidden="true"
+          >
+            {{avatar @user imageSize="large"}}
+          </a>
+        </div>
+      </div>
+      <div class="user-detail">
+        <div class="name-line">
+          <a
+            href={{userPath @user.username}}
+            data-user-card={{@user.username}}
+            aria-label={{i18n
+              "user.profile_possessive"
+              username=@user.username
+            }}
+          >
+            <span class="username">
+              {{#if this.prioritizeName}}
```

Comment:
you could use the function directly in the template - then the whole component can be template-only 😃 

```suggestion
              {{#if (prioritizeNameInUx @user.name)}}
```

Reply:
ESLint complains about template-only components:

```
7:16  error  Do not create empty backing classes for Glimmer template tag only components  ember/no-empty-glimmer-component-classes
```

I've looked at https://github.com/ember-cli/eslint-plugin-ember/blob/12a80773849273e6eee2ae3f06da31cdf76768c7/docs/rules/no-empty-glimmer-component-classes.md, but I don't see a way to make this work without in-lining the whole template in calling/parent components?

Reply:
Template-only components are declared like this:

```gjs
const MyComponent = <template>
  …
</template>

export default MyComponent;
```

As opposed to glimmer components:

```gjs
import Component from "@glimmer/component";

export default MyComponent extends Component {
  <template>
    …
  </template>
}
```

we have quite a few of template-only components in core already! The upside of those is that they're faster – there's no js class/instance backing them, so its creation is faster and their memory footprint is a bit smaller

Reply:
Ohhh, soo nice! Done!

Reply:
I'm curious though, are template-only components a standard practice/feature in Ember apps? Why doesn't the [documentation ](https://github.com/ember-cli/eslint-plugin-ember/blob/12a80773849273e6eee2ae3f06da31cdf76768c7/docs/rules/no-empty-glimmer-component-classes.md) for the `ember/no-empty-glimmer-component-classes`  rule mention how to create template-only components?

Reply:
It is a standard practice now, and the doc kinda mentions how to do this – with a caveat.

> To fix violations of this rule:
> In apps: Remove the backing class entirely until it is actually needed.

It was written before gjs was a thing, so it focuses on cases where you have a hbs template and a component that is just this:

```js
import Component from '@glimmer/component';

class MyComponent extends Component {}
```

^ and that is exactly what the gjs form transpiles into, an empty component – with a template set on it. (hence the lint error) 😃

Reply:
Aha, I see, thanks for the explanation!